### PR TITLE
feat(copyright): Use package based dependency for json

### LINKED ISSUE #1084: Change Json HPP dependency to package-based
This works then Deb9 based and related systems as the package is there.

For red hat / yum based systems, there needs to be another solution.

### PR DESCRIPTION
**Pull request by GMishx**
## Description

Remove dependency from external file download for json-hpp and shift to packaged based dependency (`libjsoncpp` from Debian based and `jsoncpp` from RedHat based systems).

### Changes

1. Remove download for external json-hpp file.
1. Add `jsoncpp` package dependency.

## How to test

Compile the `copyright` package in Debian and RedHat based systems and check the JSON output.

```bash
$ sudo ./utils/fo-installdeps -ye
$ make copyright
$ cd src/copyright/agent
$ ./copyright -J ./copyright.cc
```

Closes #1084

### CODE REVIEW

**Comment by mcjaeger** [comment]
appreciate this commit, but I did some testing on 3.4.0 rc1 already to make sure it is working. I am not so sure about adding this on short final. maybe tests fail because of timing issues of course.

**Comment by GMishx** [comment]
The tests are now fixed. Can you please check again?

**Comment by mcjaeger** [comment]
(from discussion) let#s compare output of copyright with open ssl for external dependency and package dependency.

**Comment by mcjaeger** [comment]
(from discussion:) better to use the c++ lib instead of the c one

**Comment by GMishx** [comment]
Found `jsoncpp` for Fedora 27.

And can be enabled in CentOS 7 using EPEL.
`sudo yum install epel-release`

File: src/copyright/agent/copyright.cc [hunk 1]
```diff
@@ -54,7 +54,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 #include "copyright.hpp"
 
 #ifndef DISABLE_JSON
```
Comment (ag4ums) [inline src/copyright/agent/copyright.cc]: we are installing the lib now so "#ifndef DISABLE_JSON"  can be removed,

File: src/copyright/mod_deps [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      case "$CODENAME" in
+        jessie|trusty)
+          apt-get $YesOpt install libjsoncpp0;;
+        stretch|buster|sid|artful|bionic|xenial|cosmic)
+          apt-get $YesOpt install libjsoncpp1;;
+        *) echo "ERROR: Unknown or Unsupported $DISTRO $CODENAME release, please report to the mailing list"; exit 1;;
+      esac;;
+    Fedora)
+      yum $YesOpt install \
+        jsoncpp-devel
```
Comment (ag4ums) [inline src/copyright/mod_deps]: need to be change to runtime lib

File: src/copyright/agent/copyright.cc (hunk 1 above)
Comment (GMishx) [reply src/copyright/agent/copyright.cc]: Done

File: src/copyright/mod_deps (hunk 2 above)
Comment (GMishx) [reply src/copyright/mod_deps]: Done

**Comment by GMishx** [comment]
Tested on Ubuntu 18.04 and CentOS 7

**Comment by GMishx** [comment]
Compilation failed on Fedora 27 and fixed accordingly.

**Comment by mcjaeger** [comment]
tested again with u1604: works

**Pull request merged**